index.bs

```diff
@@ -2,16 +2,715 @@
+--------------------------------------------
+
+### Definition ### {#module-interaction-definition}
+
+[=Local end definition=]:
+
+<pre class="cddl local-cddl">
+
+InteractionEvent = (InteractionCapturedOutputEvent)
+
+</pre>
+
+### Types ### {#module-interaction-types}
+
+<pre class="cddl local-cddl">
+InteractionCapturedOutputParameters = {
+  data: text,
+  *text => any
+}
+</pre>
+
+### Commands ### {#module-interaction-commands}
+
+#### The interaction.sendKeys Command #### {#module-interaction-keydown}
+
+The <dfn>interaction.sendKeys</dfn> command simulates typing on a keyboard.
+
+Note: This is intended to be similar to WebDriver's <a href="https://w3c.github.io/webdriver/#element-send-keys">Element Send Keys</a>. [[WEBDRIVER]]
+
+<dl>
+  <dt>Command Type
+  <dd>
+
+  <pre class="cddl remote-cddl">
+  InteractionSendKeysCommand = {
+    method: "interaction.sendKeys",
+    params: InteractionSendKeysParameters
+  }
+
+  InteractionSendKeysParameters = {
+    "text" => text,
+    *text => any
+  }
+  </pre>
+
+  <dt>Return Type
+  <dd>
+
+  <pre class="cddl">
+  EmptyResult
+  </pre>
+
+</dl>
+
+The [=remote end steps=] given |session| and |command parameters| are:
+
+1. [=Try=] to [=check that one of the expected applications has focus=].
+2. ...
+
+### Events ### {#module-interaction-events}
+
+<dl>
+  <dt>Event Type
+  <dd>
+
+  <pre class="cddl local-cddl">
+  InteractionCapturedOutputEvent = {
+    method: "interaction.capturedOutput",
+    params: InteractionCapuredOutputParameters
+  }
+  </pre>
+
+</dl>
+
+The [=remote end event trigger=] is:
+
+When the assistive technology would send some text |data| (a string, without speech-specific markup or annotations) to the Text-To-Speech system, or equivalent for non-speech assistive technology software, run these steps:
+
+1. Optionally, return. This step allows for an [=implementation-defined=] security check, to sandbox what information to expose.
+2. Let |params| be a [=map=] matching the `InteractionCapturedOutputParameters` production with the `data` field set to |data|.
+3. Let |body| be a [=map=] matching the `InteractionCapturedOutputEvent` production with the `params` field set to |params|.
+4. [=Emit an event=] with |session| and |body|.
+
+The actions Module {#module-actions}
+------------------------------------
+
+### Definition ### {#module-actions-definition}
+
+[=Local end definition=]:
+
+<pre class="cddl local-cddl">
+
+</pre>
+
+### Types ### {#module-actions-types}
+
+<pre class="cddl local-cddl">
+
+</pre>
+
+### Commands ### {#module-actions-commands}
+
+#### The actions.perform Command #### {#module-actions-perform}
+
+The <dfn>actions.perform</dfn> command simulates user intput actions such as key down and key up on a keyboard.
+
+Note: This is intended to be similar to WebDriver's <a href="https://w3c.github.io/webdriver/#perform-actions">Perform Actions</a>. [[WEBDRIVER]]
+
+<dl>
+  <dt>Command Type
+  <dd>
+
+  <pre class="cddl remote-cddl">
```

Comment:
By my reading of [rfc8610](https://datatracker.ietf.org/doc/html/rfc8610), the proposed syntax is valid CDDL, and if that's so, then the Node.js package's behavior is a deficiency.

Addressing the deficiency by limiting this proposal's usage of CDDL sounds like the lowest-friction solution. At the same time, doing so would hide a rough edge for future work since contributors will not necessarily recognize that the full language is not available to them, and they may not be willing/able to deviate from their structure of choice in order to satisfy a tool. It may be better to extend the utility or find an alternative that's more complete. That depends on factors which are beyond my perspective at the moment, so I'm going to hold off on this for the time being.

Reply:
I've worked around the bug for now and filed an issue: https://github.com/saucelabs/cddl/issues/7